### PR TITLE
parser: fix bug for parsing builtin UTC_TIME/UTC_TIMESTAMP/CUR_TIME/CURRENT_TIME/CURRENT_TIMESTAMP

### PR DESCRIPTION
**Pull request by spongedu**
### The first bug:
In mysql: builtin function `UTC_TIME`,`UTC_TIMESTAMP`,`CUR_TIME`, `CURRENT_TIMESTAMP`, `CURRENT_TIME` accepts 0 or 1 parameter. In the latter case, the parameter **must be unsigned int** , or there would be an parse error.  No such check was done in TiDB currently. TiDB may fail when optimizer infer types sometimes, and pass some other time(depends on whether the actual parameter  can be cast to type 'int'), which is not expected. examples:
**In Mysql**
```sql
mysql> select utc_time(c_int) from t;
ERROR 1064 (42000): You have an error in your SQL syntax; check the manual that corresponds to your MySQL server version for the right syntax
 to use near 'c_int) from t' at line 1
mysql> select utc_time(null) from t;
ERROR 1064 (42000): You have an error in your SQL syntax; check the manual that corresponds to your MySQL server version for the right syntax
 to use near 'null) from t' at line 1
mysql> select utc_time(-1) from t;
ERROR 1064 (42000): You have an error in your SQL syntax; check the manual that corresponds to your MySQL server version for the right syntax
 to use near '-1) from t' at line 1
mysql> select utc_time(1.0) from t;
ERROR 1064 (42000): You have an error in your SQL syntax; check the manual that corresponds to your MySQL server version for the right syntax
 to use near '1.0) from t' at line 1
mysql> select utc_time("4") from t;
ERROR 1064 (42000): You have an error in your SQL syntax; check the manual that corresponds to your MySQL server version for the right syntax
 to use near '"4") from t' at line 1
```

**In TiDB:**
```sql
tidb> select utc_time(c_int) from t;
ERROR 1105 (HY000): cannot convert <nil>(type <nil>) to int64
tidb> select utc_time(null);
ERROR 1105 (HY000): cannot convert <nil>(type <nil>) to int64
tidb> select utc_time(-1);
ERROR 1105 (HY000): cannot convert <nil>(type <nil>) to int64
tidb> select utc_time(1.0);
Field   1:  `utc_time(1.0)`
Catalog:    `def`
Database:   ``
Table:      ``
Org_table:  ``
Type:       TIME
Collation:  binary (63)
Length:     0
Max_length: 10
Decimals:   1
Flags:      BINARY 


+---------------+
| utc_time(1.0) |
+---------------+
| 16:03:43.3    |
+---------------+
1 row in set (0.01 sec)
tidb> select utc_time("1");
Field   1:  `utc_time("1")`
Catalog:    `def`
Database:   ``
Table:      ``
Org_table:  ``
Type:       TIME
Collation:  binary (63)
Length:     0
Max_length: 10
Decimals:   1
Flags:      BINARY 


+---------------+
| utc_time("1") |
+---------------+
| 16:04:13.4    |
+---------------+
1 row in set (0.00 sec)

```

### The second bug:
Just like `UTC_TIMESTAMP`, `UTC_TIME`  support 'No parameter, No brackets' from in mysql. Which is not supported in TiDB for now.

**In Mysql**
```sql
mysql> select utc_time;
Field   1:  `utc_time`
Catalog:    `def`
Database:   ``
Table:      ``
Org_table:  ``
Type:       TIME
Collation:  binary (63)
Length:     8
Max_length: 8
Decimals:   0
Flags:      NOT_NULL BINARY 

+----------+
| utc_time |
+----------+
| 15:58:54 |
+----------+
1 row in set (0.00 sec)

```
**In TiDB:**
```sql
tidb> select utc_time;
ERROR 1054 (42S22): Unknown column 'utc_time' in 'field list'

```

### CODE REVIEW

**Comment by zz-jason** [comment]
@breeswish @XuHuaiyu PTAL

File: parser/parser.y [hunk 1]
```diff
@@ -2431,7 +2431,7 @@ ReservedKeyword:
 "ADD" | "ALL" | "ALTER" | "ANALYZE" | "AND" | "AS" | "ASC" | "BETWEEN" | "BIGINT"
 | "BINARY" | "BLOB" | "BOTH" | "BY" | "CASCADE" | "CASE" | "CHANGE" | "CHARACTER" | "CHECK" | "COLLATE"
 | "COLUMN" | "CONSTRAINT" | "CONVERT" | "CREATE" | "CROSS" | "CURRENT_DATE" | "CURRENT_TIME"
-| "CURRENT_TIMESTAMP" | "CURRENT_USER" | "DATABASE" | "DATABASES" | "DAY_HOUR" | "DAY_MICROSECOND"
+| "CURRENT_TIMESTAMP" | "CUR_TIME"| "CURRENT_USER" | "DATABASE" | "DATABASES" | "DAY_HOUR" | "DAY_MICROSECOND"
```
Comment (breeswish) [inline parser/parser.y]: CUR_TIME is not reserved keywords according to doc: https://dev.mysql.com/doc/refman/5.7/en/keywords.html

Comment (spongedu) [inline parser/parser.y]: @breeswish Yep. It's wrong here. `CUR_TIME` is not a keyword. `CURTIME` must be called with brackets. The parameter, if have one, must be constant unsigned int. I'll fix it. thx for your review.

**Comment by spongedu** [comment]
@breeswish Fix `CUR_TIME` related issue. PTAL.

**Comment by breeswish** [comment]
LGTM

**Pull request merged**